pkg/controller/argocd/rolebinding.go

```diff
@@ -124,19 +126,22 @@ func (r *ReconcileArgoCD) reconcileRoleBinding(name string, rules []v1.PolicyRul
 		Name:     role.Name,
 	}
 
-	if name == applicationController {
-		if err := applyReconcilerHook(cr, roleBinding, ""); err != nil {
-			return err
-		}
-	}
-
 	controllerutil.SetControllerReference(cr, roleBinding, r.scheme)
 	if roleBindingExists {
 		if name == dexServer && isDexDisabled() {
 			// Delete any existing RoleBinding created for Dex
 			return r.client.Delete(context.TODO(), roleBinding)
 		}
-		return r.client.Update(context.TODO(), roleBinding)
+
+		// if the RoleRef changes, delete the existing role binding and create a new one
+		if !reflect.DeepEqual(roleBinding.RoleRef, existingRoleBinding.RoleRef) {
+			if err = r.client.Delete(context.TODO(), existingRoleBinding); err != nil {
+				return err
+			}
+		} else {
+			existingRoleBinding.Subjects = roleBinding.Subjects
+			return r.client.Update(context.TODO(), existingRoleBinding)
```

Comment:
If there is no changes, we should update it.

Reply:
if there is no change, k8s api would handle it and won't update anything !! if there's a change it'll update the object.

Reply:
This logic looks odd.  We treat RoleRef differently.  We delete/re-recreate existingRoleBinding if RoleRef has changed but we just re-assign Subjects and Update the existingRoleBinding.   Why is the difference?